spring-boot-project/spring-boot-docs/src/main/asciidoc/howto.adoc

```diff
@@ -2108,26 +2116,69 @@ See {spring-batch-api}/core/configuration/annotation/EnableBatchProcessing.html[
 
-[TIP]
-.Specifying job parameters on the command line
-====
-Unlike command line option arguments that <<spring-boot-features.adoc#boot-features-external-config-command-line-args,set properties in the `Environment`>> (i.e. by starting with `--`, such as `--my-property=value`), job parameters have to be specified on the command line without dashes (e.g. `jobParam=value`).
-====
-
-If the application context includes a `JobRegistry`, the jobs in `spring.batch.job.names` are looked up in the registry instead of being autowired from the context.
-This is a common pattern with more complex systems, where multiple jobs are defined in child contexts and registered centrally.
-
 See {spring-boot-autoconfigure-module-code}/batch/BatchAutoConfiguration.java[BatchAutoConfiguration] and https://github.com/spring-projects/spring-batch/blob/master/spring-batch-core/src/main/java/org/springframework/batch/core/configuration/annotation/EnableBatchProcessing.java[@EnableBatchProcessing] for more details.
 
+[[howto-spring-batch-running-command-line]]
+=== Running from the Command Line
+
+Running Spring Batch with Spring Boot from the command line differs from running Spring
+Batch by itself from the command line. Spring Boot uses a class called
+`JobLaunchingCommandLineRunner`. Spring Batch uses a class called `CommandLineJobRunner`.
+They work a bit differently. However, for the most part, the differences do not cause
+trouble. However, they parse command line arguments differently, which can cause trouble,
+as described in the next section.
+// TODO What are the other differences between the two?
+
+==== Passing Command-line Arguments
+
+If you run Spring Batch with Spring Boot, Spring Boot strips the first `-` character (a
+hyphen) from each command line argument. For example, `--exampleArgument` becomes
+`-exampleArgument`. Whether a command-line option has one hyphen or two often makes no
+difference in Spring Boot. However, in Spring Batch, putting a single `-` character
+before the `jobParameters` parameter indicates that Spring Batch should not use the
+`jobParameters` value as the identifier for the `Job`. Best practice is to use the
+`jobParameters` value as the identifier for the `Job`, so this issue may cause problems.
+To avoid the issue, you should generally use no `-` characters for the command-line
+options that you pass to Spring Boot on behalf of Spring Batch, as shown in the following
+example:
+
+[source]
+jobParameters="someParameter"
+
+However, if you mean to not use the `jobParameters` value as the identifier for the
+`Job`, use two hyphens, as shown in the following example:
+
+[source]
+--jobParameters="someParameter"
+
+In the second example, Spring Boot passes the parameter to Spring Batch as
+`-jobParameters="someParameter"`, and "someParameter" does not become the name of the
+`Job`.
+// TODO In that case, what is the name of the job?
```

Comment:
I must admit I am bit confused by the current phrasing. Shouldn't we rephrase it around stating that Spring Boot uses `--` as a signal for application arguments and that this class with the way Spring Batch core works?

Reply:
For this TODO, it is not the name of the job that is relevant, it is rather the job instance identity. When there is "-" in front of a parameter, the parameter is considered as non-identifying and will not contribute to the identity of a the job instance. So the sentence could be re-phrased to something like:

`In the second example, Spring Boot passes the parameter to Spring Batch as
 `-someParameter="someValue"`, and "someParameter" will be considered as a non-identifying job parameter.`

Reply:
I wrote a paragraph at the top of this section to address Stephane's concern. Then I made the change Mahmoud suggested.